app/src/main/java/com/github/onedirection/Event.java

```diff
@@ -0,0 +1,81 @@
+package com.github.onedirection;
+
+import java.util.Objects;
+
+class Event {
+
+    final private int id;
+    final private String name;
+    final private String location;
+    final private String date;
+    final private String start_time;
+    final private String end_time;
+
+
+    public Event(int id,String name, String location, String date, String start_time, String end_time){
+
+        this.name = Objects.requireNonNull(name);
+        this.location = Objects.requireNonNull(location);
+        this.date = Objects.requireNonNull(date);
+        this.start_time = Objects.requireNonNull(start_time);
+        this.end_time = Objects.requireNonNull(end_time);
+        this.id = id;
+
+    }
+
+    public Event set_name(String new_value){

```

Comment:
Java methods' shouldn't use snake_case.
```suggestion
    public Event setSame(String new_value){
```
Same for all setters.